src/App.tsx

```diff
@@ -15,49 +16,139 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  reverse = 'reverse',
+}
+
+interface FilterParams {
+  sortMethod: string | '';
+  reverseMethod: string;
+}
+
+function getSortedGoods(
+  goods: string[],
+  {
+    sortMethod,
+    reverseMethod,
+  }: FilterParams,
+) {
+  const sortedGoods = [...goods];
+
+  if (reverseMethod && sortMethod === SortType.alphabet) {
+    return sortedGoods
+      .sort((good1, good2) => good1.localeCompare(good2))
+      .reverse();
+  }
+
+  if (reverseMethod && sortMethod === SortType.length) {
+    return sortedGoods
+      .sort((good1, good2) => good1.length - good2.length)
+      .reverse();
+  }
+
+  if (reverseMethod) {
+    return sortedGoods.reverse();
+  }
+
+  if (sortMethod) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortMethod) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortMethod, setSortMethod] = useState('');
+  const [reverseMethod, setreverseMethod] = useState('');
```

Comment:
it's better to use `boolean` value for such things
Also, it will be useful: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig